worlds/_bizhawk/__init__.py

```diff
@@ -160,6 +160,16 @@ async def get_hash(ctx: BizHawkContext) -> str:
     return res["value"]
 
 
+async def get_memory_size(ctx: BizHawkContext, domain: str) -> int:
+    """Gets the size in bytes of the specified memory domain"""
+    res = (await send_requests(ctx, [{"type": "MEMORY_SIZE", "domain": domain}]))[0]
+
+    if res["type"] != "MEMORY_SIZE_RESPONSE":
+        raise SyncError(f"Expected response of type MEMORY_SIZE but got {res['type']}")
```

Comment:
```suggestion
        raise SyncError(f"Expected response of type MEMORY_SIZE_RESPONSE but got {res['type']}")
```
This is what would follow the pattern of the other nearby functions.

It seems like it would be better to pass an `expected_response: str` to the `send_requests` function so that this check can be done inside `send_requests`, for less repetition in all of these wrapping functions.
But that probably doesn't belong in this PR.

Reply:
Good catch.

I definitely see the duplication. I suppose `send_requests` should be able to check that the responses "follow the rules" so to speak. Just passing the expected type wouldn't be right, since you can chain as many types of requests as you want. I'll give it some thought.